agenta-web/src/lib/helpers/utils.ts

```diff
@@ -26,25 +35,44 @@ export const EvaluationTypeLabels: Record<EvaluationType, string> = {
     [EvaluationType.auto_webhook_test]: "Webhook Test",
 }
 
-export const saveOpenAIKey = (key: string) => {
+export const saveLlmProviderKey = (providerName: string, keyValue: string) => {
     if (typeof window !== "undefined") {
         // TODO: add encryption here
-        localStorage.setItem(openAItoken, key)
+        const keys = JSON.parse(localStorage.getItem(llmAvailableProvidersToken) ?? "{}")
+        keys[providerName] = keyValue
+        localStorage.setItem(llmAvailableProvidersToken, JSON.stringify(keys))
     }
 }
 
-export const getOpenAIKey = (): string => {
+export const getLlmProviderKeys = () => {
     // precedence order: local storage, env variable, empty string
-    let key
+    let keys: {[key: string]: string} = {}
+    for (const prov of llmAvailableProviders) {
+        keys[prov] = ""
+    }
+
+    if (typeof window !== "undefined") {
+        keys = JSON.parse(localStorage.getItem(llmAvailableProvidersToken) ?? "{}")
+    }
+    // the "NEXT_PUBLIC_OPENIA_API_KEY logic here must be rethought
+    // return key ?? process.env.NEXT_PUBLIC_OPENAI_API_KEY ?? ""
+    return keys ?? {}
+}
+
+export const getSingleLlmProviderKey = (providerName: string): string => {
+    let key = ""
```

Comment:
I would implement this function like this:

```javascript
return getLlmProviderKeys(providerKey)
```

Reply:
I reviewed this change and made sense, but I also needed a function that could return me the provider keys, so I made the change and created a specific function for what I needed